client/src/components/organisms/RequestGroupForm.tsx

```diff
@@ -0,0 +1,351 @@
+import RichTextField from '../atoms/RichTextField'
+import { RootState } from '../../data/reducers'
+import { TagInput } from '../atoms/TagInput'
+import { TextField } from '../atoms/TextField'
+import { upsertRequestGroup } from '../../data/actions'
+
+interface StateProps {
+  requestGroups: Array<RequestGroup>
+}
+
+interface DispatchProps {
+  upsertRequestGroup: typeof upsertRequestGroup
+}
+
+
+type Props = StateProps
+  & DispatchProps
+  & {
+    handleClose: () => void,
+    requestGroupId?: string,
+    operation: "create" | "edit"
+  };
+
+const RequestGroupForm: FunctionComponent<Props> = (props: Props) => {
+  const [initialRequestGroup, setInitialRequestGroup] = useState<RequestGroup | null>(null)
+  const [showAlertDialog, setShowAlertDialog] = useState(false)
+  const [changeMade, setChangeMade] = useState(false)
+  const [name, setName] = useState("")
+  const [description, setDescription] = useState("")
+  const [image, setImage] = useState("")
+  const [requestTypesToIdMap, setRequestTypesToIdMap] = useState(new Map<string, string>())
+  const [requestTypeNames, setRequestTypeNames] = useState<Array<string>>([])
+  const [nameError, setNameError] = useState("")
+  const [descriptionError, setDescriptionError] = useState("")
+  const [imageError, setImageError] = useState("")
+  const [requestTypesError, setRequestTypesError] = useState("")
+
+  const images = ["https://source.unsplash.com/RcgiSN482VI",
+    "https://source.unsplash.com/7ydep8OEvbc",
+    "https://source.unsplash.com/0hiUWSi7jvs",
+    "https://source.unsplash.com/uWVWQ8gF8PE",
+    "https://source.unsplash.com/ZQ_0jk66-1E",
+    "https://source.unsplash.com/7EWCIzU4TIg",
+    "https://source.unsplash.com/FOrNjVWdUzM",
+  ]
+
+
+  const requestGroupQuery = gql`
+  query FetchRequestGroup($id: ID!) {
+    requestGroup(id: $id) {
+      id
+      name
+      description
+      image
+      requestTypes {
+        id
+        name
+        deleted
+      }
+    }
+  }`
+
+  if (props.requestGroupId) {
+    useQuery(requestGroupQuery, {
+      variables: { id: props.requestGroupId, },
+      onCompleted: (data: { requestGroup: RequestGroup }) => {
+        const res: RequestGroup = JSON.parse(JSON.stringify(data.requestGroup)); // deep-copy since data object is frozen
+
+        setInitialRequestGroup(res)
+        setName(res.name ? res.name : "")
+        setDescription(res.description ? res.description : "")
+        setImage(res.image ? res.image : "")
+
+        res.requestTypes?.forEach((requestType) => {
+          if (requestType.name && requestType._id) {
+            setRequestTypesToIdMap(requestTypesToIdMap.set(requestType.name, requestType._id))
+          }
+        })
+
+        setRequestTypeNames(
+          res.requestTypes ?
+            res.requestTypes
+              .filter((requestType) => requestType.name)
+              .map((requestType) => requestType.name as string)
+            : []
+        )
+      }
+    });
+  }
+
```

Comment:
Nitpick, organize all these methods by which field of requestGroup they work on (e.g. requestType stuff all together)

Reply:
will do